### PR TITLE
Fix symbolic pass for handling `IntrinsicElementalFunction` in `print()`

### LINKED ISSUE #2457: `print()` messes up with integer returned by list.pop()
```
def f():
    a :list[i32]= [1, 2, 3, 4,]
    print(a.pop(),a)
    
f()

(lp) C:\Users\kunni\lpython>python try.py
4 [1, 2, 3]

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py
[1, 2, 3, 4]
```

### PR DESCRIPTION
**Pull request by kmr-srbh**
fixes #2457 

```python
from lpython import i32

def f():
    a :list[i32]= [1, 2, 3, 4,]
    print(a.pop(),a)
    
f()
```
```console
(lp) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py
4 [1, 2, 3]
```

### CODE REVIEW

**Comment by kmr-srbh** [comment]
@Shaikh-Ubaid could you please suggest some more testcases?

File: src/libasr/pass/replace_symbolic.cpp [hunk 1]
```diff
@@ -847,9 +849,10 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                 } else {
                     print_tmp.push_back(val);
                 }
-            } else {
-                print_tmp.push_back(x.m_values[i]);
-            }
+            } 
+            // else {
+            //     print_tmp.push_back(x.m_values[i]);
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/replace_symbolic.cpp:854]: Why is this removed @kmr-srbh ?

**Comment by Shaikh-Ubaid** [comment]
```python
from lpython import i32

def f():
    a :list[i32]= [1, 2, 3, 4,]
    print(a.pop(),a)
    
f()
```

Can you add this exact test case?

File: src/libasr/pass/replace_symbolic.cpp (hunk 1 above)
Comment (kmr-srbh) [reply src/libasr/pass/replace_symbolic.cpp:854]: Oh no! My bad. It must have got uncommented by mistake. I am reverting it.

File: src/libasr/pass/replace_symbolic.cpp [hunk 2]
```diff
@@ -847,7 +849,8 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                 } else {
                     print_tmp.push_back(val);
                 }
-            } else {
+            } 
+            else {
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/replace_symbolic.cpp:853]: Please revert this.

File: integration_tests/test_intrinsic_function_mixed_print.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    print("1 is located at:", my_list.index(1))
+    my_list.append(2)
+    print("2 is present", my_list.count(2), "times")
+    print(my_list.pop(), my_list)
+
+    # dict and dict methods
+    my_dict: dict[str, i32] = {"first": 1, "second": 2, "third": 3}
+    print("Keys:", my_dict.keys())
+    print("Value of 'third':", my_dict.pop("third"))
+
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_intrinsic_function_mixed_print.py:16]: Can you also add asserts in this test and also enable it in integration_tests? Its good to have this in both places integration_tests and reference runtime output.

**Comment by Shaikh-Ubaid** [comment]
Please mark as "Ready for review" when ready.

**Comment by kmr-srbh** [comment]
@Shaikh-Ubaid could you please look into the reason for the failing checks? They seem unrelated to the PR.

File: integration_tests/test_intrinsic_function_mixed_print.py (hunk 3 above)
Comment (Shaikh-Ubaid) [reply integration_tests/test_intrinsic_function_mixed_print.py:16]: Do you mind adding appropriate asserts wherever possible in the test?

Comment (kmr-srbh) [reply integration_tests/test_intrinsic_function_mixed_print.py:16]: @Shaikh-Ubaid what would we assert? Aren't we only printing here?

Comment (Shaikh-Ubaid) [reply integration_tests/test_intrinsic_function_mixed_print.py:16]: Ok. You can assert on the value that got popped and the list that is left after popping.

Comment (kmr-srbh) [reply integration_tests/test_intrinsic_function_mixed_print.py:24]: Done.

**Review by Shaikh-Ubaid — APPROVED** [review]
Perfect! Thanks. It looks great!

**Pull request merged**